### ISSUE TITLE
[Feature request] New backend based on PaddleOCR-VL-For-Manga

### ISSUE THREAD

**Issue by Gilfaro** [AuroraWright/owocr#50]
I know that getting paddle is a a lot of work to integrate but as you can check on the demo the model works pretty well for both manga and non manga local ocr for vertical and horizontal text. Currently I would say the best general use local model when it comes to accuracy.
[huggingface demo](https://huggingface.co/spaces/jzhang533/paddleocr-vl-for-manga-demo)

sample code taken from author to run on cpu, for gpu you would want the commented torch.bfloat16 and flash_attention_2 installed
from tests it only works on transformers==4.55
```
from pathlib import Path
import torch
from PIL import Image
from tqdm import tqdm
from transformers import AutoModelForCausalLM, AutoProcessor

DEVICE = "cuda" if torch.cuda.is_available() else "cpu"

CHOSEN_TASK = "ocr"  # Options: 'ocr' | 'table' | 'chart' | 'formula'
PROMPTS = {
    "ocr": "OCR:",
    "table": "Table Recognition:",
    "chart": "Chart Recognition:",
    "formula": "Formula Recognition:",
}

model_path = "jzhang533/PaddleOCR-VL-For-Manga"

model = (
    AutoModelForCausalLM.from_pretrained(
        model_path,
        trust_remote_code=True,
        torch_dtype= torch.float # for gpu torch.bfloat16,
        #attn_implementation="flash_attention_2",
    )
    .to(DEVICE)
    .eval()
)

processor = AutoProcessor.from_pretrained(
    model_path, trust_remote_code=True, use_fast=True
)

folder_path = Path("./images")

for item_path in tqdm(sorted(folder_path.iterdir()), desc="Processing images"):
    image = Image.open(item_path).convert("RGB")

    messages = [
        {
            "role": "user",
            "content": [
                {"type": "image", "image": image},
                {"type": "text", "text": PROMPTS[CHOSEN_TASK]},
            ],
        }
    ]

    text = processor.apply_chat_template(
        messages, tokenize=False, add_generation_prompt=True
    )

    inputs = processor(text=[text], images=[image], return_tensors="pt")
    inputs = {
        k: (v.to(DEVICE) if isinstance(v, torch.Tensor) else v)
        for k, v in inputs.items()
    }

    with torch.inference_mode():
        generated = model.generate(
            **inputs,
            max_new_tokens=256,
            do_sample=False,
            use_cache=True,
        )

    input_length = inputs["input_ids"].shape[1]
    generated_tokens = generated[:, input_length:]  # Slice only new tokens
    answer = processor.batch_decode(generated_tokens, skip_special_tokens=True)[0]

    print()
    print(item_path)
    print(answer)
```
